### PR TITLE
dev-libs/spdlog: bump 1.5.0

### PR DESCRIPTION
**Pull request by stkw0**

### CODE REVIEW

**Comment by candrews** [comment]
spdlog 1.4.2 fails to compile with `dev-libs/libfmt-6.1.2` (which is the current latest version):
```
/var/tmp/portage/dev-libs/spdlog-1.4.2/work/spdlog-1.4.2/include/spdlog/details/thread_pool-inl.h: In member function ‘bool spdlog::details::thread_pool::process_next_msg_()’:
/var/tmp/portage/dev-libs/spdlog-1.4.2/work/spdlog-1.4.2/include/spdlog/details/thread_pool-inl.h:119:9: error: ‘assert’ was not declared in this scope
  119 |         assert(false && "Unexpected async_msg_type");
      |         ^~~~~~
compilation terminated due to -Wfatal-errors.
```
[build.log](https://github.com/gentoo/gentoo/files/3998935/build.log)

I believe spdlog is missing at least one `#include <cassert>` statement.

**Comment by stkw0** [comment]
Fixed @candrews :)

**Comment by candrews** [comment]
There are conflicts... if you fix the conflicts I shall happily merge this PR.

Thanks again!

File: dev-libs/spdlog/spdlog-1.4.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake-utils
```
Comment (a17r) [inline dev-libs/spdlog/spdlog-1.4.2.ebuild]: One reason for conflict is the port to `cmake.eclass` of older ebuilds, yet this bump is still outdated.

File: dev-libs/spdlog/spdlog-1.4.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/gabime/${PN}"
+else
+	SRC_URI="https://github.com/gabime/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64"
+fi
+
+LICENSE="MIT"
+SLOT="0/1"
+IUSE="static-libs test"
```
Comment (a17r) [inline dev-libs/spdlog/spdlog-1.4.2.ebuild]: `static-libs` is new, is there any reason to have that? If not, please drop it.

Comment (stkw0) [reply dev-libs/spdlog/spdlog-1.4.2.ebuild]: It was added in the PR made by @lovesegfault, I don't know if he has some special reason, at the moment I don't see any so I dropped it as requested.

Comment (lovesegfault) [reply dev-libs/spdlog/spdlog-1.4.2.ebuild]: I'm fine with dropping it :)

File: dev-libs/spdlog/spdlog-1.5.0.ebuild [hunk 3]
```diff
@@ -21,19 +21,29 @@ SLOT="0/1"
...
 DEPEND="
-	>=dev-libs/libfmt-5.0.0
+	>=dev-libs/libfmt-5.3.0:=
 "
 RDEPEND="${DEPEND}"
 
-PATCHES=( "${FILESDIR}/${PN}-1.0.0-unbundle-fmt.patch" )
+#PATCHES=(
+#	"${FILESDIR}/${PN}-1.4.2-fix-compilation.patch"
+#)
```
Comment (candrews) [inline dev-libs/spdlog/spdlog-1.5.0.ebuild]: Why is this commented out?

If it's not applicable, I should be removed (and not left as a comment)

File: dev-libs/spdlog/spdlog-1.5.0.ebuild [hunk 4]
```diff
@@ -21,19 +21,29 @@ SLOT="0/1"
...
+src_prepare() {
+	cmake_src_prepare
+	rm -r include/spdlog/fmt/bundled || die "Failed to delete bundled libfmt"
+}
 
+src_configure() {
 	local mycmakeargs=(
-		-DSPDLOG_BUILD_EXAMPLES=no
 		-DSPDLOG_BUILD_BENCH=no
+		-DSPDLOG_BUILD_EXAMPLE=no
```
Comment (candrews) [inline dev-libs/spdlog/spdlog-1.5.0.ebuild]: Should this be `EXAMPLES` or `EXAMPLE`?

I personally don't know, but I want to make sure this change is correct and not a mistake.

Comment (stkw0) [reply dev-libs/spdlog/spdlog-1.5.0.ebuild]: yes, it's correct. From CMakeLists.txt `option(SPDLOG_BUILD_EXAMPLE "Build example" ${SPDLOG_MASTER_PROJECT})`

File: dev-libs/spdlog/spdlog-1.5.0.ebuild (hunk 3 above)
Comment (stkw0) [reply dev-libs/spdlog/spdlog-1.5.0.ebuild]: Sorry, I was testing if that patch was still needed and forgot to remove the comment.

**Pull request closed without merging**